libsolidity/ast/AST.h

```diff
@@ -1386,6 +1386,10 @@ class TryStatement: public Statement
 	Expression const& externalCall() const { return *m_externalCall; }
 	std::vector<ASTPointer<TryCatchClause>> const& clauses() const { return m_clauses; }
 
+	ASTPointer<TryCatchClause> successClause() const;
```

Comment:
Since ownership stays with the AST, I think a plain pointer would be better here.

Reply:
I like. thanks!